### ISSUE TITLE
Improvements to the DatabaseBackend interface

### ISSUE THREAD

**Issue by aaclayton** [foundryvtt/foundryvtt#10214]
*This issue is still a WIP and is evolving*

## Overall
- Substantially improved documentation regarding the parameters of database operations

## Get
- `index` and `indexFields` are not properties of the `operation` rather than properties of `operation.options`

**Comment by JPMeehan** [comment]
Could we get more documentation on DatabaseGetOperation#query?

**Comment by aaclayton** [comment]
> Could we get more documentation on DatabaseGetOperation#query?

Yeah - do you mind helping out by creating a separate issue for this documentation request?

**Comment by JPMeehan** [comment]
Done with #10477
